README.md

```diff
@@ -14,10 +14,74 @@ For installation instructions, please see our [install manual][install]
 
 ## Set up your Dev environment
 
-1. Make sure you have Postgresql installed
-1. Clone this repo
-1. Run `bin/setup`
-1. Start the app with `foreman start`
+1. Make sure you have PostgreSQL and Redis installed
+2. Clone this repo
+3. Run `bin/setup`
+4. Start the app with `foreman start`
+
+## Set up your Dev environment using Docker
+
+You can also set up your local development environment without having to install
+any outside dependencies. For this, use the Docker Compose configuration that's
+available in this repository. This configuration automatically mounts your
+local code inside a container, so that you can make changes and commit your
+code as if you were running a local development server without Docker.
+
+First, install Docker for your workstation: https://www.docker.com/community-edition
+
+Then, use `docker-compose` commands to set up a local environment:
+
+```
+$ docker-compose run --rm web /bin/bash
+# rails db:create
+# rails db:schema:load
+```
+
+Optionally, run the seeds to already have an initial user and if you don't
+need to test the onboarding:
+
+```
+# rails db:seed
+```
+
+Open up another tab in your terminal, or exit the shell session inside the
+container again via Control-D.
+
+Then, start the local development server:
+
+```
+$ docker-compose up
+```
+
+### Running tests
+
+You can run tests via:
+
+```
+$ docker-compose run --rm web /bin/bash
```

Comment:
Hm, the problem with this is that it becomes really hard to remember. I feel that (especially in tdd) you need to quickly go from making changes to running your tests. I don't feel this helps with that. Do you think there are different options we can take with this?

Reply:
@jvanbaarsen You can just keep a shell session open with this right? It's the same as having a terminal tab open that you use to run the tests without using Docker.

Reply:
So it attached to the container with this command and keeps the terminal session running?

Reply:
@jvanbaarsen Exactly. So all the things you normally want to do in a command line session when working on your Rails app are executed in that container. For example:

- migrating the database
- running tests
- generating controllers/models

Reply:
aaaaaah ok, makes sense 👍 Maybe we can even make a binstub for it like: `bin/enter-container` or something so you don't need to remember this command

Reply:
@jvanbaarsen Yeah, a utility script like that would be a great convenience.